src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/DacMethodsUsageAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Acuminator.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Acuminator.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DacMethodsUsageAnalyzer : PXDiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create
+            (
+                Descriptors.PX1031_DacCannotContainInstanceMethods,
+                Descriptors.PX1032_DacPropertyCannotContainMethodInvocations
+            );
+
+        internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+        {
+            compilationStartContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeDacDeclaration(syntaxContext, pxContext), SyntaxKind.ClassDeclaration);
+        }
+
+        private void AnalyzeDacDeclaration(SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            ClassDeclarationSyntax classDeclaration = syntaxContext.Node as ClassDeclarationSyntax;
+            INamedTypeSymbol typeSymbol = syntaxContext.SemanticModel.GetDeclaredSymbol(classDeclaration, syntaxContext.CancellationToken);
+
+            if (typeSymbol != null && IsDac(typeSymbol, pxContext))
+            {
+                IEnumerable<MethodDeclarationSyntax> dacMethods = classDeclaration.DescendantNodes().OfType<MethodDeclarationSyntax>();
```

Comment:
@metallmonster, the dacMethods are not used

Reply:
@SENya1990 yes, this is remain of refactoring, thanks, I'll fix it